CRM/Utils/File.php

```diff
@@ -753,6 +757,7 @@ public static function findFiles($dir, $pattern, $relative = FALSE) {
       : '@' . preg_quote(DIRECTORY_SEPARATOR) . '\.@';
 
     $dir = rtrim($dir, '/');
+    $baseDepth = substr_count($dir, '/');
```

Comment:
If this PR does proceed, this would give a random number between 0, and, say, 16, on windows. On a typical install civicrm root looks like `C:/path/to/drupal\sites\all\modules\civicrm\`, and of course sometimes people edit it.

Reply:
@demeritcowboy Yeah, I had meant to ping you about that. :)

I guess the only way out is through. Pushed a change that should work with both delimiters (while still keeping the string-operations minimal).

Reply:
Erm - sorry I wasn't trying to open a discussion about configuring the search list - my understanding is this IS existing functionality....

```
/**
 * Which directories should we exclude when scanning the codebase for things
 * like extension .info files, or .html partials or .xml files etc. This needs
 * to be a valid preg_match() pattern.
 *
 * If you do not define it, a pattern that excludes dirs starting with a dot is
 * used, e.g. to exclude .git/). Adding suitable patterns here can vastly speed
 * up your container rebuilds and cache flushes. The pattern is matched against
 * the absolute path. Remember to use your system's DIRECTORY_SEPARATOR the
 * examples below assume /
 *
 * The default excludes node_modules (can be huge), various CiviCRM dirs that
 * are unlikely to have anything we need to scan inside, and (what could be
 * your) Drupal's private file storage area. It does not exclude
 * vendor but you are likely to see an improvement by adding it.
 *
 * See https://docs.civicrm.org/sysadmin/en/latest/setup/optimizations/#exclude-dirs-that-do-not-need-to-be-scanned
 * and also discussion on including vendor (excluded) in https://lab.civicrm.org/dev/core/-/issues/2031
 */
if (strtoupper(substr(PHP_OS, 0, 3)) !== 'WIN' && !defined('CIVICRM_EXCLUDE_DIRS_PATTERN')) {
  define('CIVICRM_EXCLUDE_DIRS_PATTERN', '@/(\.|node_modules|js/|css/|bower_components|packages/|sites/default/files/private)@');
}
```

Reply:
@artfulrobot contributed that I believe